src/BlockTypes/Cart.php

```diff
@@ -49,8 +49,12 @@ public function render( $attributes = array(), $content = '' ) {
 		$data_registry = Package::container()->get(
 			\Automattic\WooCommerce\Blocks\Assets\AssetDataRegistry::class
 		);
-		$data_registry->add( 'shippingCountries', WC()->countries->get_shipping_countries() );
-		$data_registry->add( 'shippingStates', WC()->countries->get_shipping_country_states() );
+		if ( ! $data_registry->exists( 'shippingCountries' ) ) {
+			$data_registry->add( 'shippingCountries', WC()->countries->get_shipping_countries() );
```

Comment:
I wonder if we could add a method combining both to reduce the code here? Not sure what it would be called though :D

Reply:
Yeah... I was wondering the same. I like the current implementation because each method does just one thing, but if we see this pattern used over and over again (checking if a setting exists, and then adding it) it might make sense creating a method (or a util function) to do both.

I still thing we can defer it in case we see this is used in several other blocks. For now it's only _Cart_ and _Checkout_ so I think it's ok as-is.

Reply:
it does make sense to have the same function, I think `add` should do that (add if not exist)

Reply:
We more or less discussed that in #1751. I recognize I don't have a strong opinion about this. The issue with `add` is that the consumer would not know if it overrided the previous value or it was ignored, so we thought that an explicit method to check for a setting existence would avoid possible confusion.

I think we can work on that in a follow-up if we think that's something that needs to be fixed.